### ISSUE TITLE
[kafka] - Permission denied when creating config files

### ISSUE THREAD

**Issue by henro001** [bitnami/charts#7736]
**Which chart**:
kafka 7.3/11.8

**Describe the bug**

After Kuberentes upgrade 1.19 to 1.20 kafka stopped working the pods wont start with these errors

```
cp: cannot create regular file '/opt/bitnami/kafka/conf/./connect-console-sink.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./connect-console-source.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./connect-distributed.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./connect-file-sink.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./connect-file-source.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./connect-log4j.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./connect-mirror-maker.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./connect-standalone.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./consumer.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./log4j.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./producer.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./tools-log4j.properties': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./trogdor.conf': Permission denied
cp: cannot create regular file '/opt/bitnami/kafka/conf/./zookeeper.properties': Permission denied
11.8 
/opt/bitnami/scripts/libfile.sh: line 34: /opt/bitnami/kafka/config/server.properties: Permission denied
```

**To Reproduce**

Install Kafka helm chart with default values

**Expected behavior**
Pods start.

**Version of Helm and Kubernetes**:
Helm 3.5.2
Kube 1.20.11

**Additional context**
Dockerfile:
https://github.com/bitnami/bitnami-docker-kafka/blob/master/2/debian-10/Dockerfile

These containers by default run with UID 1001. GID not set.
 /opt/bitnami/ is owned by root:root 775

`drwxrwxr-x 1 root root 4096 Mar  5  2020 /opt/bitnami/`

During the startup there are tasks to copy/move the config files between directories in /opt/bitnami/. If the container runs with UID 1001, GID unset, this copy process will fail unless GID is 0. 

I observed the following when I run the image

Kube 1.20
`uid=1001 gid=1(daemon) groups=1(daemon),1001`

Kube 1.19
`uid=1001 gid=0(root) groups=0(root),1(daemon),1001`

Docker
`uid=1001 gid=0(root) groups=0(root)`

For some reason in 1.19 or in docker the gid is 0, but in 1.20  GID is 1, which is the reason for permission denied.

Concerns: Does this image rely on GID 0? This could be sub optimal for security. Could this be addressed by changing the owner of /opt/bitnami to match the UID?
If this is intentional, what is the proper way to configure the chart?

Side note: this probably affects other bitnami charts, I am looking at zookeper and postgres and see same permission denied error

**Comment by henro001** [comment]
and just a side note, this probably affects other bitnami charts, I am looking and zookeper and postgres and see same permission denied error

**Comment by henro001** [comment]
I fixed this by chowning the directory in the dockerfile. Both kafka and zookeeper now start successfully.
chown -R 1001:1001 /opt/bitnami/kafka
Should the owner be 1001?

**Comment by henro001** [comment]
I think I found what changed across the upgrade to 1.20 but I will need to find some more info on this.
We use podsecurity policies in all our environments. ( I am aware there are big changes coming re podsecuritypolicy).
We dont create explicit policy for kafka/zookeeper so they are running with `restricted`. With restricted there is a supplementary group 1 added to running pods by default. GID 0 is default when no GID is specified.
This has changed after upgrade and GID 0 is no longer present. I thought it was just one cluster but I confirmed this with second lab cluster that has recently been upgraded to 1.20. So my problem is the podsecurity policy in combination with upgrade to 1.20. I think i have two options: 
1. chown in initContainer
2. create custom podsecurity policy to allow GID 0 and then add supplementalGroup 0
